src/resp.rs

```diff
@@ -24,25 +24,28 @@ impl<T : Send + 'static> SingleResponse<T> {
         SingleResponse(Box::new(f))
     }
 
-    pub fn metadata_and_future<F>(metadata: Metadata, result: F) -> SingleResponse<T>
+    pub fn metadata_and_future<F>(metadata: Metadata, result: F, trailing: GrpcFutureSend<Metadata>) -> SingleResponse<T>
```

Comment:
I think it is better to have additional function like `metadata_and_future_and_trailing_metadata`, keeping this function as is for convenience.